### ISSUE TITLE
Green AVI videos in Video for Windows 1.1e under Windows for Workgroups 3.11 

### ISSUE THREAD

**Issue by delsydebothom** [joncampbell123/dosbox-x#3815]
### Describe the bug

Any video played in Media Player in WFW 3.11 comes out green using VFW 1.1e. The same problem is happening in Microsoft Home products that embed AVI videos. The emulated video card is the S3 Trio64 V+, and the video driver I have installed is Version 1.70.04, S3 Flat Model Windows Trio765 Driver, dated 09/22/97.

![krnl386_000](https://user-images.githubusercontent.com/117696415/200450699-a8cb9264-d413-4ca5-9da5-9b4cd0da914a.png)

### Steps to reproduce the behaviour

Install WFW 3.11 from floppy disks via USB drive
Move device drivers from host system into a new DRIVERS directory created in the DOSBox-X folder mounted as C:\.
Install S3 video drivers.
Install Soundblaster 16 drivers
Install VFW 1.1e
Play .avi file  

### Expected behavior

Since Quicktime and MPEG videos play fine in their respective players, and colors are repoduced accurately in everything else, it seems like this ought to work correctly as well.

### What operating system(s) this bug have occurred on?

Windows 10

### What version(s) of DOSBox-X have this bug?

64-bit SDL1 DOSBox-X 2022.09.0

### Used configuration

```ini
#
language                  = 
title                     = 
fastbioslogo              = false
startbanner               = true
bannercolortheme          = default
dpi aware                 = auto
quit warning              = auto
working directory option  = default
working directory default = 
show advanced options     = false
resolve config path       = true
hostkey                   = mapper
mapper send key           = ctrlaltdel
ime                       = auto
synchronize time          = false
machine                   = svga_s3trio64v+
captures                  = capture
autosave                  = 
saveslot                  = 1
savefile                  = 
saveremark                = true
forceloadstate            = false
a20                       = mask
memsize                   = 64
nocachedir                = false
freesizecap               = cap
convertdrivefat           = true
```

### Output log

```text
LOG: 2482041375 ERROR INT10:Function 6F00 not supported
LOG: pixratio 1.000, dw false, dh false
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: DEBUG: INT 2Fh Windows 286/386 DOSX init broadcast issued (ES:BX=0000:0000 DS:SI=0000:0000 CX=0000 DX=0000 DI=030b(aka version 3.11))
LOG:  [Enhanced mode]
LOG:
LOG: 2495884505 ERROR CPU:Illegal Unhandled Interrupt Called 68
LOG: 2497610717 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: 2497611006 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: 2497611263 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: 2497611392 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: 2497611521 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: 2497611650 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: 2497611779 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: 2497611908 ERROR CPU:Illegal Unhandled Interrupt Called 5C
LOG: DEBUG: INT 2Fh Windows virtual device 'PAGEFILE' callout (BX(deviceID)=0x0021 CX(function)=0x0000)
LOG: DEBUG: INT 2Fh Windows virtual device 'VMD' callout (BX(deviceID)=0x000c CX(function)=0x0001)
LOG: INT 15h mouse initialized to 3-byte protocol
LOG: INT 15h mouse reset
LOG: INT 15h reporting mouse device ID 0x00
LOG: DEBUG: INT 2Fh Windows virtual device 'VNETBIOS' callout (BX(deviceID)=0x0014 CX(function)=0x0000)
LOG: DEBUG: INT 2Fh Windows virtual device 'DOSMGR' callout (BX(deviceID)=0x0015 CX(function)=0x0000)
LOG: DEBUG: INT 2Fh Windows virtual device 'DOSMGR' callout (BX(deviceID)=0x0015 CX(function)=0x0004)
LOG: DEBUG: INT 2Fh Windows virtual device 'DOSMGR' callout (BX(deviceID)=0x0015 CX(function)=0x0001)
LOG: DEBUG: INT 2Fh Windows virtual device 'DOSMGR' callout (BX(deviceID)=0x0015 CX(function)=0x0003)
LOG: DEBUG: INT 2Fh Windows virtual device 'DOSMGR' callout (BX(deviceID)=0x0015 CX(function)=0x0005)
LOG: DEBUG: INT 2Fh Windows virtual device 'PAGEFILE' callout (BX(deviceID)=0x0021 CX(function)=0x0000)
LOG: PNP OS changed I/O read port to 0x273 (from 0x20B)
LOG: DEBUG: INT 2Fh Windows virtual device 'V86MMGR' callout (BX(deviceID)=0x0006 CX(function)=0x0000)
LOG: DEBUG: INT 2Fh Windows virtual device 'VMD' callout (BX(deviceID)=0x000c CX(function)=0x0000)
LOG: pixratio 1.200, dw false, dh true
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: 2520348289 ERROR MOUSE:Unhandled videomode 69 on reset
LOG: 2520384314 ERROR INT10:Function 7F00 not supported
LOG: pixratio 1.000, dw false, dh false
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: INFO: switched to full screen mode
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: XGA: Wrote to port ff00 with fffffff8, len 4
LOG: INFO: switched to window mode
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: INFO: switched to full screen mode
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: INFO: switched to window mode
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: Screen report: Method 'Win98base' (1366.000 x 768.000 pixels) at (0.000 x 0.000) (361.421 x 203.200 mm) (14.229 x 8.000 in) (96.000 x 96.000 DPI)
```

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by Torinde** [comment]
What was the solution?

**Comment by rderooy** [comment]
Based on what was posted on Vogons, it was to re-install the S3 video driver.

Apparently when you install the S3 video driver, it copies 2 more files if it finds that VFW is installed. Without those files you get the green screen.

**Comment by delsydebothom** [comment]
Ah, my apologies for not saying anything. I ended up re-installing the drivers (just by copying over the old ones), but what finally ended up working was to install a modified version of VFW 1.1 modified for ELSA cards. Why did that work? No idea. If I had seen the suggestion about UDH.dll before I got it working again, I'd have completely unloaded the S3 drivers and installed them from scratch. Maybe that also would have worked?
